### PR TITLE
Dashboard

### PR DESCRIPTION
**Pull request by Soimil**

### CODE REVIEW

File: apps/ui/lib/ui_web/live/price_chart_live.ex [hunk 1]
```diff
@@ -95,3 +100,5 @@ defmodule UiWeb.PriceChartLive do
     }
   end
 end
+
+14 / (14 + 32) * 100
```
Comment (Cinderella-Man) [inline apps/ui/lib/ui_web/live/price_chart_live.ex]: what's this?

File: apps/ui/lib/ui_web/live/gaining_losing_trades_live.ex [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+    %{
+      :chart_data => gaining_losing_data(from, to),
+      :symbols => symbols
+    }
+  end
+
+  defp gaining_losing_data(from, to) do
+    rows_data = Hefty.Trades.count_gaining_losing(from, to).rows
+
+    rows = rows_data
```
Comment (Cinderella-Man) [inline apps/ui/lib/ui_web/live/gaining_losing_trades_live.ex]: all of this logic should be inside `count_gaining_losing` function

File: apps/ui/lib/ui_web/live/gaining_losing_trades_live.ex [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+        <!-- /.footer -->
+      </div>
+    """
+  end
+
+  def mount(%{}, socket) do
+    [from, to] = Hefty.Utils.Datetime.get_timestamps(T.today())
+
+    symbols =
+      Hefty.Trades.fetch_trading_symbols(from, to)
```
Comment (Cinderella-Man) [inline apps/ui/lib/ui_web/live/gaining_losing_trades_live.ex:69]: why only trading symbols? I should be able to select any symbol

btw. I would assume that fetching trading symbols would query `naive_trader_settings` for a list of `enabled`

File: apps/ui/lib/ui_web/live/gaining_losing_trades_live.ex [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+      |> Enum.map(&List.to_string(&1))
+
+    symbol =
+      symbols
+      |> List.first()
+
+    {:ok, assign(socket, data: get_data(symbols), symbol: symbol)}
+  end
+
+  def handle_info(%{event: "trade_event"}, socket) do
```
Comment (Cinderella-Man) [inline apps/ui/lib/ui_web/live/gaining_losing_trades_live.ex]: this widget is not dependent on trade events?

File: apps/ui/lib/ui_web/live/gaining_losing_trades_live.ex [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+    rows = rows_data
+    |> Enum.group_by(fn [head | _tail] -> head end)
+    |> Enum.map(&row_builder(&1))
+
+    %{
+      :rows => rows,
+    }
+  end
+
+  defp row_builder(row) do
```
Comment (Cinderella-Man) [inline apps/ui/lib/ui_web/live/gaining_losing_trades_live.ex]: as above - should be part of `count_gaining_losing`

**Review by Cinderella-Man — CHANGES_REQUESTED** [review]

**Comment by Cinderella-Man** [comment]
Thanks @Soimil for your PR :+1:  there's few improvements that I flagged in comments - could you please fix those?

File: apps/ui/lib/ui_web/live/gaining_losing_trades_live.ex (hunk 3 above)
Comment (Soimil) [reply apps/ui/lib/ui_web/live/gaining_losing_trades_live.ex:69]: gets all symbols that occur in trading within a specified period of time

File: apps/ui/lib/ui_web/live/price_chart_live.ex (hunk 1 above)
Comment (Soimil) [reply apps/ui/lib/ui_web/live/price_chart_live.ex]: Please, make it clearer. Never mind, I know now.

File: apps/ui/lib/ui_web/live/gaining_losing_trades_live.ex (hunk 4 above)
Comment (Soimil) [reply apps/ui/lib/ui_web/live/gaining_losing_trades_live.ex]: Sorry, already clean up

File: apps/ui/lib/ui_web/live/gaining_losing_trades_live.ex (hunk 5 above)
Comment (Soimil) [reply apps/ui/lib/ui_web/live/gaining_losing_trades_live.ex]: Thank you for suggestion.

File: apps/ui/lib/ui_web/live/gaining_losing_trades_live.ex (hunk 2 above)
Comment (Soimil) [reply apps/ui/lib/ui_web/live/gaining_losing_trades_live.ex]: Thx, already there

**Review by Soimil — COMMENTED** [review]
No problem. I will do this asap. Thanks @Cinderella-Man for all suggestions.

File: apps/hefty/lib/hefty/trades.ex [hunk 6]
```diff
@@ -166,4 +166,27 @@ defmodule Hefty.Trades do
...
+    query =
+      "SELECT symbol, (cast(profit_base_currency as double precision) > 0) AS gaining, " <>
+      "COUNT(*) FROM trades " <>
+      "WHERE sell_time >= #{from} AND " <>
+      "sell_time < #{to} " <>
+      "GROUP BY (symbol, cast(profit_base_currency as double precision) > 0);"
+
+    Ecto.Adapters.SQL.query!(Hefty.Repo, query).rows
+      |> Enum.group_by(fn [head | _tail] -> head end)
+      |> Enum.map(fn {symbol, [[_, _, losing], [_, _, gaining]]} -> %{String.to_atom("#{symbol}") => [gaining, losing]} end)
```
Comment (Cinderella-Man) [inline apps/hefty/lib/hefty/trades.ex:180]: Instead of using `String.to_atom("#{symbol}")` it's popular to use `:"#{symbol}"`

**Review by Cinderella-Man — APPROVED** [review]

**Pull request merged**